geonode/resource/manager.py

```diff
@@ -78,24 +105,116 @@ def search(self, filter: dict, /, type: object = None) -> QuerySet:
         return _resources_queryset
 
     def exists(self, uuid: str, /, instance: ResourceBase = None) -> bool:
-        instance = instance or ResourceManager._get_instance(uuid)
-        if instance:
-            return self._resource_manager.exists(uuid, instance=instance)
+        _resource = instance or ResourceManager._get_instance(uuid)
+        if _resource:
+            return self._resource_manager.exists(uuid, instance=_resource)
         return False
 
     @transaction.atomic
     def delete(self, uuid: str, /, instance: ResourceBase = None) -> int:
-        instance = instance or ResourceManager._get_instance(uuid)
-        if instance:
-            self._resource_manager.delete(uuid, instance=instance)
-            return instance.delete()
+        _resource = instance or ResourceManager._get_instance(uuid)
+        if _resource:
+            try:
+                self._resource_manager.delete(uuid, instance=_resource)
+                _resource.get_real_instance().delete()
+                return 1
+            except Exception as e:
+                logger.exception(e)
         return 0
 
-    def create(self, uuid: str, /, instance: ResourceBase = None) -> int:
-        pass
+    @transaction.atomic
+    def create(self, uuid: str, /, type: object = None, defaults: dict = {}) -> ResourceBase:
```

Comment:
thinking out loud: could be useful to pass the same variables in both update and create? Because, if i want to pass some specific keywords or thesauri,atm it looks complicated

Reply:
actually I would like to force to perform a create and and update separately